daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Anf.scala

```diff
@@ -194,132 +258,161 @@ private[lf] object Anf {
-
-        case SEAppGeneral(func, args) =>
-          atomizeExp(
-            depth,
-            env,
-            func, {
-              case (depth, func) =>
-                atomizeExps(
-                  depth,
-                  env,
-                  args.toList, {
-                    case (depth, args) =>
-                      val func1 = makeRelativeA(depth)(func)
-                      val args1 = args.map(makeRelativeA(depth))
-                      k(depth, SEAppAtomic(func1, args1.toArray))
-                  }
-                )
-            }
-          )
-        case SEMakeClo(fvs0, arity, body0) =>
-          val fvs = fvs0.map((loc) => makeRelativeL(depth)(makeAbsoluteL(env, loc)))
-          val body = flattenToAnf(body0).wrapped
-          k(depth, SEMakeClo(fvs, arity, body))
-
-        case SECase(scrut, alts0) =>
-          atomizeExp(depth, env, scrut, {
-            case (depth, scrut) =>
-              val scrut1 = makeRelativeA(depth)(scrut)
-              val alts = flattenAlts(depth, env, alts0)
-              k(depth, SECaseAtomic(scrut1, alts))
-          })
-
-        case SELet(rhss, body) =>
-          val expanded = expandMultiLet(rhss.toList, body)
-          transformExp(depth, env, expanded, k)
-
-        case SELet1General(rhs, body) =>
-          transformLet1(depth, env, rhs, body, k)
-
-        case SECatch(body0, handler0, fin0) =>
-          flattenExp(
-            depth,
-            env,
-            body0,
-            body => {
-              flattenExp(depth, env, handler0, handler => {
-                flattenExp(depth, env, fin0, fin => {
-                  k(depth, SECatch(body.wrapped, handler.wrapped, fin.wrapped))
-                })
-              })
-            }
-          )
-
-        case SELocation(loc, body) =>
-          transformExp(depth, env, body, {
-            case (depth, body) =>
-              k(depth, SELocation(loc, body))
-          })
-
-        case SELabelClosure(label, exp) =>
-          transformExp(depth, env, exp, {
-            case (depth, exp) =>
-              k(depth, SELabelClosure(label, exp))
-          })
-
-        case x: SEAbs => throw CompilationError(s"flatten: unexpected: $x")
-        case x: SEWronglyTypeContractId => throw CompilationError(s"flatten: unexpected: $x")
-        case x: SEVar => throw CompilationError(s"flatten: unexpected: $x")
-
-        case x: SEAppAtomicGeneral => throw CompilationError(s"flatten: unexpected: $x")
-        case x: SEAppAtomicSaturatedBuiltin => throw CompilationError(s"flatten: unexpected: $x")
-        case x: SELet1Builtin => throw CompilationError(s"flatten: unexpected: $x")
-        case x: SECaseAtomic => throw CompilationError(s"flatten: unexpected: $x")
-
-    })
-
-  private[this] def atomizeExps(
+  private[this] def transformExp[A](
+      depth: DepthA,
+      env: Env,
+      exp: SExpr,
+      transform: Tx[SExpr, A],
+      k: K[AExpr, A]): Trampoline[A] =
```

Comment:
You may find the formatting becomes much nicer if you split `k` into its own parameter list (in all cases).

This would become:

```scala
  private[this] def transformExp[A](
      depth: DepthA,
      env: Env,
      exp: SExpr,
      transform: Tx[SExpr, A],
  )(k: K[AExpr, A]): Trampoline[A]
```

And its call site in `transformLet1` would look something like this:

```scala
      Bounce { () =>
        transformExp(depth1, env1, body, transform) { body1 =>
          Bounce(() => txK(AExpr(SELet1(rhs, body1.wrapped))))
        }
      }
```

Reply:
@SamirTalwar-DA Thanks, I might try this fix, because the current formatting is truly awful.